### PR TITLE
STORM-515: Clojure documentation and examples

### PR DESCRIPTION
**Pull request by brymaven**
Add some more examples with Clojure, which show using parameters with config and global grouping.
Also add tests for all clojure bolts.

### CODE REVIEW

**Comment by revans2** [comment]
+1 looks good to me.

File: examples/storm-starter/src/clj/org/apache/storm/starter/clj/bolts.clj [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+           (log-debug "Received tick tuple, triggering emit of current window counts")
+           (.markAsModified tracker)
+           (doseq [[obj count] counts]
+             (emit-bolt! collector [obj count actual-window-length])))
+         (do
+           (.incrementCount counter word)
+           (ack! collector tuple)))))))
+
+(defmacro update-rankings [& body]
+  `(if (TupleUtils/isTick ~'tuple)
```
Comment (knusbaum) [inline examples/storm-starter/src/clj/org/apache/storm/starter/clj/bolts.clj]: Relying on symbols from the expansion environment is sort of a bad idea. It does make the call look cleaner, but it leaves potential for accidental breakage in mysterious ways.

File: examples/storm-starter/test/clj/org/apache/storm/starter/clj/bolts_test.clj [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+     (.prepare bolt# {} nil (OutputCollector.
+                       (reify IOutputCollector
+                         (emit [_ _ _ tuple#]
+                           (swap! tuples# conj tuple#))
+                         (ack [_ input]))))
+     (if (vector? ~tuples)
+       (doseq [t# ~tuples]
+         (.execute bolt# t#))
+       (.execute bolt# ~tuples))
+     (let [~'tuples @tuples#]
```
Comment (knusbaum) [inline examples/storm-starter/test/clj/org/apache/storm/starter/clj/bolts_test.clj]: `~'tuples` evaluates to `tuples` This makes the macro unhygienic.

**Comment by knusbaum** [comment]
Looks good for the most part.

**Comment by brymaven** [comment]
Good suggestions. I fixed the macros.

File: examples/storm-starter/src/clj/org/apache/storm/starter/clj/bolts.clj [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+             (emit-bolt! collector [obj count actual-window-length])))
+         (do
+           (.incrementCount counter word)
+           (ack! collector tuple)))))))
+
+(defmacro update-rankings [tuple & body]
+  `(if (TupleUtils/isTick ~tuple)
+     (do
+       (log-debug "Received tick tuple, triggering emit of current rankings")
+       (emit-bolt! ~'collector [(.copy ~'rankings)])
```
Comment (knusbaum) [inline examples/storm-starter/src/clj/org/apache/storm/starter/clj/bolts.clj]: My previous comment applies to `collector` and `rankings` as well

**Comment by knusbaum** [comment]
+1
Nice.

**Pull request merged**